examples/client.c

```diff
@@ -426,11 +426,24 @@ message_handler(struct coap_context_t *ctx,
       block_opt = coap_check_option(received, COAP_OPTION_BLOCK1, &opt_iter);
 
       if (block_opt) { /* handle Block1 */
-        block.szx = COAP_OPT_BLOCK_SZX(block_opt);
-        block.num = coap_opt_block_num(block_opt);
-
-        debug("found Block1, block size is %u, block nr. %u\n",
-        block.szx, block.num);
+        unsigned int szx = COAP_OPT_BLOCK_SZX(block_opt);
+        unsigned int num = coap_opt_block_num(block_opt);
+        debug("found Block1 option, block size is %u, block nr. %u\n", szx, num);
+        if (szx != block.szx) {
+          unsigned int bytes_sent = (block.num + 1) * (1 << (block.szx + 4));
+          if (bytes_sent % (1 << (szx + 4)) == 0) {
```

Comment:
Replacing the code with a bitwise and would be a premature optimization in my opinion. I applied your suggested change and compared the disassembly at `-O2` optimization level, the output is identical. The point is that the compiler already knows about this optimization and we don't have to apply it manually, the change will only make the code slightly more difficult to read for users less used to working with bitwise operations

GCC version: `gcc (Gentoo 5.4.0 p1.0, pie-0.6.5) 5.4.0`
Command line (default configure settings): `gcc -DHAVE_CONFIG_H -I. -I..    -isystem../include/coap -I../include/coap -pedantic -Wall -Wextra -Wformat-security -Winline -Wmissing-declarations -Wmissing-prototypes -Wnested-externs -Wpointer-arith -Wshadow -Wstrict-prototypes -Wswitch-default -Wswitch-enum -Wunused  -Wlogical-op -Wunused-result -std=c99 -g -O2 -fdiagnostics-color -D_GNU_SOURCE -DWITH_POSIX -MT client.o -MD -MP -MF .deps/client.Tpo -c -o client.o client.c`)